unit_testing/cli/test_cli_sct_image.py

```diff
@@ -1,21 +1,43 @@
-from pytest_console_scripts import script_runner
-import pytest
 import logging
 import os
 
+import pytest
+import numpy as np
+
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox.utils import extract_fname, sct_test_path
 from spinalcordtoolbox.scripts import sct_image
-from spinalcordtoolbox.utils.sys import sct_test_path
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.mark.script_launch_mode('subprocess')
-def test_sct_image_backwards_compat(script_runner):
-    ret = script_runner.run('sct_testing', '--function', 'sct_image')
-    logger.debug(f"{ret.stdout}")
-    logger.debug(f"{ret.stderr}")
-    assert ret.success
-    assert ret.stderr == ''
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+@pytest.mark.parametrize("path_in", ['t2/t2.nii.gz', 'dmri/dmri.nii.gz'])
+def test_sct_image_getorient(path_in):
+    """Run the CLI script and ."""
+    sct_image.main(argv=['-i', path_in, '-getorient'])
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_image_split_dmri(dmri_t_slices):
+    """Verify the output of '-split' matches reference image. Note: CLI script is run by the 'dmri_t_slices' fixture."""
+    _, filename, ext = extract_fname(dmri_t_slices[0])
+    ref = Image(f'dmri/{filename}{ext}')  # Reference image should exist inside working directory (sct_testing_data)
+    new = Image(dmri_t_slices[0])         # New image should be generated inside tmp directory
+    assert np.sum(ref.data - new.data) <= 1e-3
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_image_concat(tmp_path, dmri_t_slices, dmri_in):
+    """Run the CLI script and verify concatenated imaeg matches reference image."""
+    path_out = str(tmp_path / 'dmri_concat.nii.gz')
+    sct_image.main(argv=['-i'] + dmri_t_slices + ['-concat', 't', '-o', path_out])
+    ref = Image(dmri_in)
+    new = Image(path_out)
+    assert np.sum(ref.data - new.data) <= 1e-3
```

Comment:
> and adjust the tolerances to match, if necessary.

Funny enough, the difference between the reference image and the new image is 0.0, because process is just splitting and concatenating images. I'm not sure where the `1e-3` tolerance came from, but I think in this case it might be more reasonable to assert that there should be no difference at all.

Reply:
e4fa6c2